src/test/javascript/spec/component/learning-paths/components/learning-path-lecture-unit.component.spec.ts

```diff
@@ -109,6 +117,38 @@ describe('LearningPathLectureUnitComponent', () => {
         expect(component.lectureUnit()).toEqual(lectureUnit);
     });
 
+    it('should not set current learning object on error', async () => {
+        const completeLectureUnitSpy = jest.spyOn(lectureUnitService, 'completeLectureUnit').mockImplementationOnce(() => {});
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        component.setLearningObjectCompletion({ completed: true, lectureUnit: lectureUnit });
+
+        expect(completeLectureUnitSpy).toHaveBeenCalledWith(lectureUnit.lecture, {
+            completed: true,
+            lectureUnit: lectureUnit,
+        });
+        expect(setLearningObjectCompletionSpy).not.toHaveBeenCalled();
+    });
+
+    it('should set current learning object completion', async () => {
+        const completeLectureUnitSpy = jest
+            .spyOn(lectureUnitService, 'completeLectureUnit')
+            .mockImplementationOnce((lecture: Lecture, completionEvent: LectureUnitCompletionEvent) => (completionEvent.lectureUnit.completed = completionEvent.completed));
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        component.setLearningObjectCompletion({ completed: true, lectureUnit: lectureUnit });
+
+        expect(completeLectureUnitSpy).toHaveBeenCalledWith(lectureUnit.lecture, {
+            completed: true,
+            lectureUnit: lectureUnit,
+        });
+        expect(setLearningObjectCompletionSpy).toHaveBeenCalledWith(true);
```

Comment:
Can you check if we can be more specific here?

```suggestion
        expect(completeLectureUnitSpy).toHaveBeenCalledExactlyOnceWith(lectureUnit.lecture, {
            completed: true,
            lectureUnit: lectureUnit,
        });
        expect(setLearningObjectCompletionSpy).toHaveBeenCalledExactlyOnceWith(true);
```